palantirnet/search_api_federated_solr
# UMUS-74: Integrate search app into module

This PR adds the search app to the module with functionality for local development + production build.  It also declares a search page route and renders the search app on a site.

## To prep for testing
1. Checkout `develop` in umus site repo
1. Checkout this branch in `src/search_api_federated_solr`
1. Observe + investigate the new `react-search` directory

## To test search page on site
1. Build and refresh the lab blog site
    1. `phing build-one -Dsite.properties=conf/site/uofmhealth_labblog.properties`
    1. `phing refresh-one -Dsite.properties=conf/site/uofmhealth_labblog.properties`
1. Import config `drush @healthblog.consumer.local cim -y`
1. Log in to the site `drush @healthblog.consumer.local uli`
1. Browse to the search app route: `https://labblog.uofmhealth.local/search-app`
1. Observe the default styling
1. Interact with the app to confirm functionality

## To test theming the search app
1. See update to module [`README`](https://github.com/palantirnet/search_api_federated_solr/blob/umus-74-search-app-integration/README.md) with link to [`theming docs`](https://github.com/palantirnet/search_api_federated_solr/blob/umus-74-search-app-integration/docs/theme.md)
1. Follow steps to theme the search app (Note: eventually we'd include only one .scss/.css starter file) -- if you want to compile sass, you can install the search app dependencies and run `./node_modules/node-sass/bin/node-sass  ../../../../themes/custom/<your-theme>/sass/<your-sass-file> -o ../../../../themes/custom/<your-theme>/css` from within `./src/react-search`
1. Optional, build + theme the app for the consumer site (it is very similar to the default right now)

## To spin up local dev for search app
1. CD into the search app `cd react-search`
1. Install dependencies `yarn install`
1. Run the dev task: `yarn start`
1. Observe a new browser window opens and the search app loads
1. Touch a js + scss file and observe the app rebuilds and the browser reloads

## To test a new production build
1. Since you just touched a js + scss file you should have see updates to a production build
1. Kill the dev task `ctrl + c`
1. Run the production task: `yarn build`
1. Observe the git status for the module:
    1. You should see your updates in `./src` + `./build/static` as well as the module `libraries.yml` file which should now point to those new `./build/static` assets
1. Undo your changes
